src/components/map/Map.tsx

```diff
@@ -157,138 +107,219 @@ export const Map = ({
+        const deltaLng = (maxLng - minLng) * 0.1; // Add 10% padding to the bounds
+        const deltaLat = (maxLat - minLat) * 0.1; // Add 10% padding to the bounds
+        if (cameraRef.current) {
+          cameraRef.current.fitBounds(
+            [minLng - deltaLng, minLat - deltaLat],
+            [maxLng + deltaLng, maxLat + deltaLat],
+            0 // animationDuration in ms
+          );
+          setTimeout(() => {
+            setIsLoading(false);
+          }, 1000); // Timeout in Millisekunden anpassen je nach Bedarf
+          setIsInitialFit(false); // Verhindert zukünftige automatische Anpassungen
+        }
+      }
+    }
+  }, [locations, clusterDistance, zoomLevel, isInitialFit]);
+
+  const handleMyLocationButtonPress = () => {
+    if (refForMapView.current && locationSettings?.currentLocation) {
+      cameraRef.current.setCamera({
+        centerCoordinate: [
+          locationSettings.currentLocation.latitude,
+          locationSettings.currentLocation.longitude
+        ],
+        zoomLevel: 15,
+        animationDuration: 1000
+      });
+    }
+  };
+
   return (
     <View style={[styles.container, style]}>
-      <MapView
-        clusterColor={colors.primary}
-        clusterFontFamily="regular"
-        clusteringEnabled={clusteringEnabled}
-        radius={clusterDistance}
-        renderCluster={renderCluster}
-        initialRegion={initialRegion}
-        region={updatedRegion}
-        mapType={device.platform === 'android' ? MAP_TYPES.NONE : MAP_TYPES.STANDARD}
+      <LoadingSpinner loading={isLoading} />
+      <MapLibreGL.MapView 
+        style={styles.map}
+        styleJSON='https://tileserver-gl.smart-village.app/styles/osm-liberty/style.json'
         onPress={onMapPress}
         ref={refForMapView}
-        rotateEnabled={false}
-        showsBuildings={false}
-        showsPointsOfInterest={false}
-        showsUserLocation={showsUserLocation}
-        showsMyLocationButton={false}
-        toolbarEnabled={false}
-        style={[stylesForMap().map, mapStyle]}
-        userLocationPriority="balanced"
-        minZoom={minZoom}
-        mapPadding={{
-          top: 0,
-          right: 0,
-          bottom: -normalize(30),
-          left: 0
-        }}
-        legalLabelInsets={{
-          top: 0,
-          right: 0,
-          bottom: normalize(10),
-          left: 0
-        }}
+        onRegionDidChange={(region) => {
+          const newZoomLevel = Math.round(region.properties.zoomLevel);
+          if (!isNaN(newZoomLevel)) {
+            setZoomLevel(newZoomLevel);
+          }
+        }}  
+        showUserLocation={showsUserLocationSetting}
       >
-        <UrlTile
-          urlTemplate="https://tileserver-gl.smart-village.app/styles/osm-liberty/{z}/{x}/{y}.png"
-          shouldReplaceMapContent={device.platform === 'ios'}
+        <MapLibreGL.Camera
+          ref={cameraRef}
```

Comment:
Why do we have a camera with `MapLibreGL`?

Reply:
Camera is just the packages name for the functionality to show the desired map frame.
for exp. as you open the map you see immediately Berlin at zoom lvl 10 if the so called camera is set properly, instead of opening the map and seeing the whole world at zoom lvl 0